csharp/src/elements/Geometry/Polygon.cs

```diff
@@ -91,19 +132,163 @@ public bool Contains(Polygon polygon)
+        {
+            var scale = 1024.0;
+            var thisPath = this.ToClipperPath();
+            var intPoint = new IntPoint(point.X * scale, point.Y * scale);
+            if (Clipper.PointInPolygon(intPoint, thisPath) == 0)
+            {
+                return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Tests if all the 2D Vector3 points in the supplied list fall within or on the perimeter of this Polygon.
+        /// </summary>
+        /// <param name="points">The list of 2D Vector3 points to compare.</param>
+        /// <returns>
+        /// Returns true if any of the supplied 2D Vector3 points fall within or on the perimeter of this Polygon.
+        /// </returns>
+        public bool Covers(IList<Vector3> points)
+        {
+            var scale = 1024.0;
+            var thisPath = this.ToClipperPath();
+            foreach (Vector3 point in points)
+            {
+                var intPoint = new IntPoint(point.X * scale, point.Y * scale);
+                if (Clipper.PointInPolygon(intPoint, thisPath) == 0)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Tests if this Polygon covers the supplied Polygon. This Polygon covers the supplied Polygon if all points of the supplied Polygon fall within or on the perimeter of this Polygon.
+        /// </summary>
+        /// <param name="polygon">The Polygon to test for coverage by this Polygon.</param>
+        /// <returns>
+        /// Returns true if the supplied Polygon is contained by or coincides with the perimeter of the supplied Polygon.
         /// </returns>
         public bool Covers(Polygon polygon)
         {
+            var tolerance = 0.00001;
+            var clipper = new Clipper();
+            var solution = new List<List<IntPoint>>();
+            clipper.AddPath(this.ToClipperPath(), PolyType.ptSubject, true);
+            clipper.AddPath(polygon.ToClipperPath(), PolyType.ptClip, true);
+            clipper.Execute(ClipType.ctIntersection, solution);
+            if (solution.Count != 1)
+            {
+                return false;
+            }
+            var testPolygon = solution.First().ToPolygon();
+            if (Math.Abs(polygon.Area - testPolygon.Area) > tolerance)
+            {
+                return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Tests if this Polygon covers all of the supplied Polygons. This Polygon covers the supplied Polygons if all points of the supplied Polygons fall within or on the perimeter of this Polygon.
+        /// </summary>
+        /// <param name="polygons">The list of Polygons to compare.</param>
+        /// <returns>
+        /// Returns true if any of the supplied Polygons intersects with this Polygon.
+        /// </returns>
+        public bool Covers(IList<Polygon> polygons)
```

Comment:
As mentioned above, we shouldn't duplicate code here. All plural versions of a method should call the singular version.

Reply:
As above, originally how I had it. I changed it because perhaps I misunderstood this Slack comment:
ikeo [5:08 PM]
In looking through the changes my first instinct is to remove all the methods that just contain a loop calling the method.
This will reduce the size of the PR. It may be that we want to add these back in the future. But if we do this, I want it to be applied across the library.

Anthony Hauck [5:09 PM]
I figured that was a pattern you wouldn't like much -- I can self-contain those, but it's pretty annoying to have external loops everywhere to call the base method. (edited)

Reply:
> ...my first instinct is to remove all the **methods** that just...

That comment was meant to suggest that we should remove them from this PR entirely, and allow the user to call the function in a loop however they'd like.

Reply:
Done.

Reply:
I see you point, and that we should make it a larger pattern if we go here. I'll pull the plural methods out of the base library, stand by, should be done by the time you're starting your day.

Reply:
Done.